Source/JavaScriptCore/bytecode/Repatch.cpp

```diff
@@ -1257,10 +1276,22 @@ static InlineCacheAction tryCacheArrayPutByVal(JSGlobalObject* globalObject, Cod
     AccessGenerationResult result;
 
     {
+        GCSafeConcurrentJSLocker locker(codeBlock->m_lock, globalObject->vm());
+
         JSCell* base = baseValue.asCell();
 
-        AccessCase::AccessType accessType;
-        if (isTypedView(base->type())) {
+        RefPtr<AccessCase> newCase;
```

Comment:
Not really sure why we move them to the top `tryCacheArrayPutByVal`.

Reply:
We are aligning our code with the same form to `tryCacheArrayGetByVal` `tryCacheArrayInByVal` not to make something weird.